File: android/src/main/java/com/zero/flutter_qq_ads/event/AdErrorEvent.java
package com.zero.flutter_qq_ads.event;

import java.util.HashMap;

/**
 *  广告错误事件
 */
public class AdErrorEvent extends AdEvent {
    // 错误码
    int errCode;
    // 错误信息
    String errMsg;
    // 错误事件实体
    public AdErrorEvent(String adId, String action, int errCode, String errMsg) {
        super(adId, action);
        this.errCode = errCode;
        this.errMsg = errMsg;
    }

    /**
     * 重写 toMap 方法
     * @return 返回错误事件的map
     */
    @Override
    public HashMap<String, Object> toMap() {
        HashMap<String, Object> newMap= super.toMap();
        newMap.put("errCode",errCode);
        newMap.put("errMsg",errMsg);
        return newMap;
    }
}

File: ios/Classes/Event/AdErrorEvent.m
//
//  AdErrorEvent.m
//  flutter_qq_ads
//
//  Created by zero on 2021/8/13.
//

#import "AdErrorEvent.h"
#import "AdEventAction.h"

@implementation AdErrorEvent

- (id)initWithAdId:(NSString *)adId errCode:(NSInteger *)errCode errMsg:(NSString *)errMsg{
//    [super initWithAdId:adId andAction:action];
    self.adId=adId;
    self.action=onAdError;
    self.errCode=errCode;
    self.errMsg=errMsg;
    return self;
}

- (NSDictionary *)toMap{
    NSDictionary *data=[super toMap];
    [data setValue:_errMsg forKey:@"errMsg"];
    [data setValue:[NSNumber numberWithInteger:_errCode] forKey:@"errCode"];
    
    return data;
}
@end
